### PR TITLE
make shell job adaptor workdir configurable

### PR DESCRIPTION
**Pull request by andre-merzky**

### CODE REVIEW

File: src/saga/adaptors/shell/shell_job.py [hunk 1]
```diff
@@ -174,7 +174,17 @@ def _decode (data) :
...
+    {
+    'category'         : 'saga.adaptor.shell_job',
+    'name'             : 'base_workdir',
+    'type'             : str,
+    'default'          : "$HOME/.saga/adaptors/shell_job/",
+    'documentation'    : '''The adaptor stores job state information on the
+                          filesystem on the target resource.  This parameter
+                          specified what location should be used.''',
+    'env_variable'     : None
+    }
```
Comment (marksantcroos) [inline src/saga/adaptors/shell/shell_job.py]: What does this bit actually do? I recently added a config entry without putting it in here.

**Comment by marksantcroos** [comment]
Whats the context of this change btw?

File: src/saga/adaptors/shell/shell_job.py (hunk 1 above)
Comment (andre-merzky) [inline src/saga/adaptors/shell/shell_job.py]: Yeah, you can do without I gues, but then you have no type conversion, default value, documentation, env support... :)

**Comment by andre-merzky** [comment]
I am sorry, I can't find the email thread anymore, but there was a user for whom `$HOME/.saga` was not on a shared file system or something, so reconnecting to the job service's state failed when connecting to a different login node.

File: src/saga/adaptors/shell/shell_job.py (hunk 1 above)
Comment (marksantcroos) [inline src/saga/adaptors/shell/shell_job.py]: Ah ok, I took example from a non-adaptor usage.

Comment (marksantcroos) [inline src/saga/adaptors/shell/shell_job.py]: Does the attempt to fix it in b6675dd look correct to you?

Comment (andre-merzky) [inline src/saga/adaptors/shell/shell_job.py]: yes.

**Pull request merged**